ip_addr.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Walter Scheper <walter.scheper@gmail.com>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package rapid
+
+import (
+	"fmt"
+	"net"
+)
+
+const (
+	ipv4Len = 4
+	ipv6Len = 16
+)
+
+var (
+	ipv4Gen = SliceOfN(Byte(), ipv4Len, ipv4Len)
+	ipv6Gen = SliceOfN(Byte(), ipv6Len, ipv6Len)
+)
+
+type ipGen struct {
+	ipv6 bool
+}
+
+func (g *ipGen) String() string {
+	return fmt.Sprintf("IP(ipv6=%v)", g.ipv6)
+}
+
+func (g *ipGen) value(t *T) net.IP {
```

Comment:
Maybe it is good idea to use [`netip`](https://pkg.go.dev/net/netip)? With generics we already require Go 1.18.

Reply:
Makes sense. Done.